pedagogy_guide.md

```diff
@@ -0,0 +1,72 @@
+Thirty minutes is a good duration to aim for.
+
+### Consider the aim/objective of the note book
+Is the notebook a Tutorial to teach a concept or tool or a How To Guide to show how to perform a task?  Is the notebook intended to be self guided or taught by an instructor in a workshop?
+
+[divio.com](https://documentation.divio.com/structure/) provide a good overview of the differences between Tutorials and How-To Guides; both describe practical steps but Tutorials are designed to help users to understand concepts, whereas How-To Guides are more problem orientated and designed to solve problems that come up during working.  
+
+### Avoid Cognitive Overload 
+Avoid cognitive overload by focusing notebook content on one main task.  Break complex notebooks into several notebooks.
+Not the only or best guide but useful. https://blog.innerdrive.co.uk/4-ways-to-overcome-cognitive-overload?hs_amp=true
+Notebooks should be for a single - short - task.  This avoids cognitive overload.
+
+### Follow best practices for coding but avoid overly complicated code.
+Aim to follow best practices for coding but also avoid overly complicated code that obscures the teaching aim.  For example, there is no need to
+include code to manage errors and exceptions.
+
+Code needs to be understandable rather than efficient.  Avoiding trying to write code as a 'developer'.
+
+### Avoid long code blocks
+If a code block is so long that users cut-and-paste the code or press shift-enter then they are not learning, just copying.  The Carpentries advocate live-coding,
+where an instructor types code in real-time, correcting mistakes as they go and describing what they are doing.  This approach helps with pacing and length of content.
+
+While most useful in workshops, thinking of writing code live, even for self-guided tutorials, will help keep coding manageable and the cognitive load manageable.
+
+Short code blocks also help break tasks into understandable steps.  Think in terms of what steps do we need to understand to complete a task.
+
+### Use standard or well-established packages (unless introducing a new package)
+Aim to use packages either from the standard library or well-established packages.  Obscure packages or packages that are no longer
+maintained may introduce too many dependencies or end up breaking code.  The only exception here is if we are introducing a new package.
+
+### Use Live Coding for In-Person Tutorials
+Live coding is where an instructor types code in real-time, talking through what they are doing, instead of cuting-and-pasting or just running code blocks.  Participants follow along also typing code.  This approach has advantages for pacing.  An instructor can only go as quickly as they can type.  Students get practice typing code.  Students also get to see mistakes, error messages, and solutions.
```

Comment:
```suggestion
Live coding is where an instructor types code in real-time, talking through what they are doing, instead of cutting-and-pasting or just running code blocks.  Participants follow along also typing code.  This approach has advantages for pacing.  An instructor can only go as quickly as they can type.  Students get practice typing code.  Students also get to see mistakes, error messages, and solutions.
```